service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java

```diff
@@ -166,37 +124,42 @@ public void tearDown() {
-        var hl7TranslatedResponse = outputMessageWrapperMapper.map(
-            getGpcStructuredTaskDefinition,
-            ehrExtractContent);
+        final EhrExtractTemplateParameters ehrExtractTemplateParameters =
+            ehrExtractMapper.mapBundleToEhrFhirExtractParams(getGpcStructuredTaskDefinition, bundle);
+
+        final String ehrExtractContent = ehrExtractMapper.mapEhrExtractToXml(ehrExtractTemplateParameters);
+
+        final String hl7TranslatedResponse = outputMessageWrapperMapper.map(getGpcStructuredTaskDefinition, ehrExtractContent);
 
         assertThat(hl7TranslatedResponse).isEqualTo(expectedJsonToXmlContent);
     }
 
     private static Stream<Arguments> testValueFilePaths() {
-        return Stream.of(
-            Arguments.of(INPUT_JSON_WITH_PATHOLOGY_RECORD, OUTPUT_XML_USES_PATHOLOGY_RECORD),
-            Arguments.of(INPUT_JSON_WITH_MEDICATION_RECORD, OUTPUT_XML_USES_MEDICATION_RECORD),
-            Arguments.of(INPUT_JSON_WITH_STRUCTURED_CONSULTATIONS_INCLUDING_NARRATIVE_CONTENT,
-                OUTPUT_XML_USES_STRUCTURED_CONSULTATIONS_INCLUDING_NARRATIVE_CONTENT),
-            Arguments.of(INPUT_JSON_WITH_GENERAL_MISCELLANEOUS_CODED_OBSERVATION_CONTENT,
-                OUTPUT_XML_USES_GENERAL_MISCELLANEOUS_CODED_OBSERVATION_CONTENT),
-            Arguments.of(INPUT_JSON_WITH_BLOOD_PRESSURES, OUTPUT_XML_USES_BLOOD_PRESSURES),
-            Arguments.of(INPUT_JSON_WITH_IMMUNISATIONS, OUTPUT_XML_USES_IMMUNISATIONS),
-            Arguments.of(INPUT_JSON_WITH_DRUG_AND_NON_DRUG_ALLERGIES, OUTPUT_XML_USES_DRUG_AND_NON_DRUG_ALLERGIES),
-            Arguments.of(INPUT_JSON_WITH_DIARY_ENTRIES, OUTPUT_XML_USES_DIARY_ENTRIES),
-            Arguments.of(INPUT_JSON_WITH_REFERRALS, OUTPUT_XML_USES_REFERRALS),
-            Arguments.of(INPUT_JSON_WITH_PROBLEMS_AND_PROBLEM_LINKAGES, OUTPUT_XML_USES_PROBLEMS_AND_PROBLEM_LINKAGES),
-            Arguments.of(INPUT_JSON_WITH_NON_SNOMED_CODED_DATA_VIA_EGTON_CODES, OUTPUT_XML_USES_NON_SNOMED_CODED_DATA_VIA_EGTON_CODES)
-        );
+        return Stream.of(Arguments.of("9465701483_Dougill_full_20210119.json", "9465701483_Dougill_full_20210119.xml"),
+            Arguments.of("9465701483_Nel_full_20210119.json", "9465701483_Nel_full_20210119.xml"),
+            Arguments.of("9465701459_Nel_full_20210119.json", "9465701459_Nel_full_20210119.xml"),
+            Arguments.of("9465698679_Gainsford_full_20210119.json", "9465698679_Gainsford_full_20210119.xml"),
+            Arguments.of("9465700193_Birdi_full_20210119.json", "9465700193_Birdi_full_20210119.xml"),
+//            Arguments.of("9465701262_Meyers_full_20210119.json", "9465701262_Meyers_full_20210119.xml"),
+//            Arguments.of("9465699918_Magre_full_20210119.json", "9465699918_Magre_full_20210119.xml"),
+//            Arguments.of("9465701297_Livermore_full_20210119.json", "9465701297_Livermore_full_20210119.xml"),
```

Comment:
Commented out due to agent mapping issue in ReferralRequest

```
java.lang.NullPointerException: reference is marked non-null but is null

	at uk.nhs.adaptors.gp2gp.ehr.mapper.RequestStatementMapper.isReferenceToType(RequestStatementMapper.java:300)
	at uk.nhs.adaptors.gp2gp.ehr.mapper.RequestStatementMapper$InnerMapper.processAgent(RequestStatementMapper.java:120)
```

Reply:
From my analysis it looks like only the errors for the Meyers and Lamura data are caused by the null issue. Magre, Sajal and Livermore seem to have issues in the agent directory mappings not pre-populating the IdMapper